src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java

```diff
@@ -17,7 +17,7 @@
 	 * @param driver Set the driver as current.
 	 */
 	public synchronized void setCurrentDriver(Job2dDriver driver) {
-		currentDriver = driver;
+		currentDriver = driver instanceof DriverUsageMonitor ? driver : new DriverUsageMonitor(driver);
```

Comment:
Do not change this code. Try decorating in a subscriber (this way usage monitor would be optional, not hardcoded)